framework/Source/Operations/CBAddBlend.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  CBAddBlend.swift
+//
+//
+//  Created by Chiaote Ni on 2024/6/14.
+//
+
+import Foundation
+
+// It's the same as the AddBlend in GPUImage.
+// To create a separate class because the original design of GPUImage is not for executing under the async/await environment, and will easily cause a deadlock.
+// For more details, see here: https://www.notion.so/piccollage/await-newTexture-3962c12cc694452faeaa8210760898f4
+// Therefore, to run the processImage with async, this class override the newTextureAvailable function to achieve this goal to avoid the deadlock.
+// Also, follow the implementation of LookupFilter, I create an blendImageInput for this filter, which allow us to use the same interface in PicCollage when we migrate from using GPUImage3 to using CBVisualEffectBuiltIn
+public class CBAddBlend: BasicOperation {
+
+    public var intensity: Float = 1 { didSet { uniformSettings["intensity"] = intensity } }
+
+    public var blendImageInput: PictureInput? {
+        didSet {
+            inputTextures.removeValue(forKey: 1)
+            blendImageInput?.addTarget(self, atTargetIndex: 1)
+        }
+    }
+
+    public init() {
+        super.init(fragmentFunctionName:"addBlendFragment", numberOfInputs:2)
+        ({
+            intensity = 1
+        })()
+    }
+
+    public override func newTextureAvailable(_ texture: Texture, fromSourceIndex: UInt) {
+        if fromSourceIndex == 1 {
+            inputTextures[fromSourceIndex] = texture
+            return
+        }
+
```

Comment:
suggest:

```swift 
guard fromSourceIndex == 0 else {
       super.newTextureAvailable(texture, fromSourceIndex: fromSourceIndex)
       return
}

guard let blendImageInput, inputTextures[fromSourceIndex] == nil else {
      super.newTextureAvailable(texture, fromSourceIndex: fromSourceIndex)
      return
}

let lock = DispatchSemaphore(value: 0)
Task {
       await blendImageInput.processImage()
       super.newTextureAvailable(texture, fromSourceIndex: fromSourceIndex)
              lock.signal()
       }
lock.wait()
```

Reply:
The same as the previous [comment](https://github.com/cardinalblue/GPUImage3/pull/20#discussion_r1639281700), we can't call the super in this case

Reply:
I modified it

Reply:
We don't need to have lock for the else case.

Reply:
No problem, I'll modify with it in the next commit 👍

Reply:
Also, add timout for it

```swift
func wait(timeout: DispatchTime) 
```

Reply:
Yes, that'll be better, thanks!
How about 2 seconds, or 10 seconds at most?

Reply:
1s @@

Reply:
OK XD

Reply:
Will timout 1s make result fail very often?

Reply:
I think it's safe enough to do with 1 under this use case, but just 2 seconds make me feel more comfortable personally XD

Reply:
ok

Reply:
Or how about using 2, if you don't have strong feedback about this?

Reply:
nope